File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);



int main(void)
{
    string text = get_string("Text: ");
    int w = count_words(text);
    double L = (count_letters(text) * 100.0) / w;
    double S = (count_sentences(text) * 100.0) / w;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 0 && index < 17)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int l = 0;
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 32)
        {
            if (l == 0)
            {
                count += 2;
                l = 1;
            }
            else
            {
                count++;
            }
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 33 || text[i] == 63 || text[i] == 46)
        {
            count += 1;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string

def main():
    text = get_string("Text: ")
    W = count_words(text)
    L = (count_letters(text) * 100.0) / W
    S = (count_sentences(text) * 100.0) / W
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index > 0 and index < 17:
        print(f"Grade {index}")
    elif index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")


def count_letters(text):
    count = 0
    for i in range(len(text)):
        if ((ord(text[i]) > 64 and ord(text[i]) < 91)) or (ord(text[i]) > 96 and ord(text[i]) < 123):
            count += 1
    return count

def count_words(text):
    l = 0
    count = 0
    for i in range(len(text)):
        if ord(text[i]) == 32:
            if l == 0:
                count += 2
                l = 1
            else:
                count += 1
    return count

def count_sentences(text):
    count = 0
    for i in range(len(text)):
        if ord(text[i]) == 33 or ord(text[i]) == 63 or ord(text[i]) == 46:
            count += 1
    return count

main()